From b521d3ae5ffb6352febe3bbeed4d4d28f730658a Mon Sep 17 00:00:00 2001
From: Dmitry Ovsyanko <Dmitry.Ovsyanko@mts.ru>
Date: Sat, 27 Feb 2010 15:02:08 +0300
Subject: [PATCH] handler: draw_error_page for 'require' time errors

---
 Eludia/Content/Handler.pm |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/Eludia/Content/Handler.pm b/Eludia/Content/Handler.pm
index 6e2cc4d..6d545bb 100755
--- a/Eludia/Content/Handler.pm
+++ b/Eludia/Content/Handler.pm
@@ -275,8 +275,10 @@ sub setup_page {
 
 	call_for_role ('get_page');
 
-	require_both $page -> {type};	
+	eval { require_both $page -> {type} };
 	
+	$_REQUEST {error} ||= $@;
+		
 	$page -> {request_type} = 
 
 		$_REQUEST {__suggest} ? 'suggest' :
-- 
1.6.2.1217.gd7bc3

